command.go

```diff
@@ -25,3 +25,20 @@ func (t OnOffCommand) Execute(ctx Context, args map[string]interface{}) proto.De
 func (t OnOffCommand) Name() string {
 	return "action.devices.commands.OnOff"
 }
+
+// OpenCloseCommand defines how a function should handle this specific trait
+type OpenCloseCommand func(ctx Context, params interface{}) proto.DeviceError
+
+// Execute validates the request and calls the user defined handler for the device trait
+func (t OpenCloseCommand) Execute(ctx Context, args map[string]interface{}) proto.DeviceError {
+	// validate if our EXECUTE Request contains the "openPercent" object so the handler can actually handle something
+	if val, ok := args["openPercent"]; ok {
```

Comment:
this should validate ```openPrecent``` is a float before passing to ```t```, it should also attempt to extract ```openDirection``` and if it cant pass in ```OpenCloseTraitDirectionNone```

Reply:
done